pymatgen/core/structure.py

```diff
@@ -625,6 +625,35 @@ def add_spin_by_site(self, spins: Sequence[float]) -> None:
                 new_species[Species(sym, oxidation_state=oxi_state, spin=spin)] = occu
             site.species = Composition(new_species)
 
+    def add_interstitial(self, atom: str) -> None:
+        """Creates an interstitial defect in the structure in-place by adding the atom to the available voids.
+
+        Args:
+            atom (str): e.g. "H", "Li" etc.
+        """
+
+        def density(r, R):
+            return np.exp(-np.dot(r - R, r - R))
+
+        supercell = self.copy()
+        supercell.make_supercell([3, 3, 3])
+        divisions = np.array([10, 10, 10]) * 3
+        n = np.zeros(divisions)
+        for i in range(divisions[0]):
+            for j in range(divisions[1]):
+                for k in range(divisions[2]):
+                    f = [i / divisions[0], j / divisions[1], k / divisions[2]]
+                    r = np.dot(f, supercell.lattice.matrix)
+
+                    for R in supercell.cart_coords:
+                        if self.distance(R, r) < 4:
```

Comment:
The 4 A cutoff here might be better suited as an optional keyword argument that the user can specify, which would default to 4.